posts/2020-06-03-asynchronous-programming-made-resilient-with-mpft.adoc

```diff
@@ -36,39 +32,40 @@ Adding the `@Asynchronous` annotation to a method does two things:
+Usually when a method is called, it returns a result. However, when the `@Asynchronous` annotation is used, Fault Tolerance intervenes. When the annotated method is called, Fault Tolerance first schedules the method to run later on a different thread, then it returns a `CompletionStage` or `Future`. This isn't the `CompletionStage` or `Future` that's returned from the method, because the method hasn't run yet. It's an object that Fault Tolerance creates.
 
 image::/img/blog/FT-basic-asynchronous-execution.png[Fault Tolerance asynchronous execution flow]
 
-Sometime later, the method runs and returns a `CompletionStage` or `Future`. Now, Fault Tolerance will go and complete the `CompletionStage` or `Future` object which was returned to the caller earlier, with the value that was returned from the method. You might be wondering *_"What if the returned result is an exception? What can I do?"_*.  Read along.
+Sometime later, the method runs and returns a `CompletionStage` or `Future`.
+Now, Fault Tolerance completes the `CompletionStage` or `Future` that was returned earlier, with the value that was returned from the method. You might be wondering, *_"What if the returned result is an exception? What can I do?"_*. Read along.
 
-== `@Asynchronous` Use cases 
-So `@Asynchronous` sounds great, right? But how can you use it to make your asynchronous programs resilient? Let's go through 2 use cases:
+== `@Asynchronous` use cases
+So `@Asynchronous` sounds great, right? But how can you use it to make your asynchronous programs resilient? Let's go through two use cases:
 
-1. <<Applying-to-asynch-api, Applying Fault Tolerance to an API call which is already asynchronous>>
+1. <<Applying-to-asynch-api, Applying Fault Tolerance to an API call that is already asynchronous>>
 2. <<Running-methods-in-parallel, Using the `@Asynchronous` annotation to do multiple things in parallel with resilience>>
 
 [#Applying-to-asynch-api]
 === 1. Applying Fault Tolerance to an asynchronous API call
-One use of `@Asynchronous` is to apply Fault Tolerance to an asynchronous API call which returns a `CompletionStage`. Without `@Asynchronous`, you wouldn't be able to apply Fault Tolerance to the call. Let's see why.
+One use of `@Asynchronous` is to apply Fault Tolerance to an asynchronous API call, which returns a `CompletionStage`. Without `@Asynchronous`, you wouldn't be able to apply Fault Tolerance to the call. Let's see why.
 
-In this example, we'll use the JAX-RS client API for calling remote REST services, introduced in JAX-RS 2.1. We can build up a request to fetch a String from a given URL with a GET request and the return type is a `CompletionStage` of String:
+In this example, we'll use the JAX-RS client API, which was introduced in JAX-RS 2.1, for calling remote REST services. We can build up a request to fetch a String from a given URL with a GET request, where the return type is a `CompletionStage` of String:
```

Comment:
Technically not true.

The JAX-RS client API was introduced in JAX-RS 2.0. The reactive API extensions were added in JAX-RS 2.1.